numpy/__init__.pyi

```diff
@@ -3654,3 +3661,8 @@ class broadcast:
     def __next__(self) -> Tuple[Any, ...]: ...
     def __iter__(self: _T) -> _T: ...
     def reset(self) -> None: ...
+
+class _SupportsDLPack(Protocol[_T_contra]):
+    def __dlpack__(self, *, stream: Optional[int] = ...) -> _PyCapsule: ...
```

Comment:
```suggestion
    def __dlpack__(self, *, stream: None | _T_contra = ...) -> _PyCapsule: ...
```
The `|` operator is just a more compact shortcut for `Union` or `Optional`,
but more importantly: the typevar is missing from the argument type.

Reply:
The signature changed to 
```
def __dlpack__(self: NDArray[number[Any]], *, stream: None = ...) -> _PyCapsule: ...
```

Reply:
My comment applied (and still does) to the `_SupportsDLPack` protocol, used by the `from_dlpack` function for identifying objects implementing `__dlpack__`.

The `ndarray.__dlpack__` method itself does look perfectly fine.